sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/interfaces.scala

```diff
@@ -164,6 +164,9 @@ case class AggregateExpression(
       case _ => aggFuncStr
     }
   }
+
+  override protected def withNewChildrenInternal(newChildren: IndexedSeq[Expression]): Expression =
+    super.legacyWithNewChildren(newChildren)
```

Comment:
Why not implement this one?

Reply:
Fixed it. Probably it makes sense to have a special node with nodes with a child and an optional child. There are a number of some nodes. Manually implementing them is tricky. I'd like to ideally only have to implement this method like this:
```
override protected def withNewChildrenInternal(newChildren: IndexedSeq[Expression]): Expression = 
  copy(children = newChildren)
```
That is, only for nodes with a list of children. We'll do these tweaks in a followup.